integration/query_compat_test.go

```diff
@@ -41,6 +41,7 @@ type queryCompatTestCase struct {
 	resultType     compatTestCaseResultType // defaults to nonEmptyResult
 	resultPushdown bool                     // defaults to false
 
+	skipIDCheck   bool   // skip check collected IDs, check results only
```

Comment:
I think I need an explanation of when collected IDs should be skipped.

Reply:
I rewrote the comment to this flag to add an explanation.